src/SerialFirmata.cpp

```diff
@@ -48,13 +48,18 @@ boolean SerialFirmata::handleSysex(byte command, byte argc, byte *argv)
     Stream *serialPort;
     byte mode = argv[0] & SERIAL_MODE_MASK;
     byte portId = argv[0] & SERIAL_PORT_ID_MASK;
+    if (portId >= SERIAL_READ_ARR_LEN) return false;
 
     switch (mode) {
       case SERIAL_CONFIG:
         {
           long baud = (long)argv[1] | ((long)argv[2] << 7) | ((long)argv[3] << 14);
           serial_pins pins;
-
+#if defined(SERIAL_STORE_AND_FORWARD)    
+          lastAvailableBytes[portId] = 0;
+          lastReceive[portId] = 0;
+          maxCharDelay[portId] = 50000 / baud; // 8N1 = 10 bits per char -> 50 bits max. char/char delay 
```

Comment:
change comment to: "8N1 = 10 bits per char -> 50 bits max. 50000 = 50bits * 1000ms/s"